### ISSUE TITLE
Add support for codechef Problem URL

### ISSUE THREAD

**Issue by Rishabh-malhotraa** [Rishabh-malhotraa/caucus#43]
# Requirement
Currently, Caucus only supports codeforces and atcoder links, but it would be great to add support for CodeChef links.

- Changes need to be made in the caucus server to add the functionality to scrape the CodeChef question from the get response.
Also, some UI changes in the front-end, to tell the user that we support CodeChef links.

![image](https://user-images.githubusercontent.com/54576074/135949483-82a2c54a-96ea-4dd9-ba50-73f00178abde.png)

**Files to be modified**
- [caucus-server/src/utils/scrapeQuestion.ts](https://github.com/Rishabh-malhotraa/caucus-server/blob/main/src/utils/scrapeQuestion.ts)
- [caucus/src/component/QuestionsPane/ProblemList/ProblemList.tsx](https://github.com/Rishabh-malhotraa/caucus/blob/main/src/component/QuestionsPane/ProblemList/ProblemList.tsx)

**Comment by ujju20** [comment]
Can I work on this issue?

**Comment by Rishabh-malhotraa** [comment]
> Can I work on this issue?

Go ahead, if you need any help hmu on discord.

**Comment by ujju20** [comment]
> > Can I work on this issue?
> 
> Go ahead, if you need any help hmu on discord.

Sure

**Comment by starkkumarkk1611** [comment]
Hi,
@Rishabh-malhotraa 
CodeChef API is restricted. It means I have to do it with Pure Web scraping. I would Like to do it with npm package jsdom.

**Comment by Rishabh-malhotraa** [comment]
> Hi, @Rishabh-malhotraa CodeChef API is restricted. It means I have to do it with Pure Web scraping. I would Like to do it with npm package jsdom.

Yes, @starkkumarkk1611 CodeChef does not have a public API that gives us access to question, we need to make a get request to the problem page, parse the HTML and select only those div which encloses the questions.

You can look at how its implemented [here](https://github.com/Rishabh-malhotraa/caucus-server/blob/main/src/utils/scrapeQuestion.ts) in caucus-server. 

We use [cheerio.js](https://cheerio.js.org/)  for parsing and traversing the dom elements

**Comment by starkkumarkk1611** [comment]
Thanks I will give my to implement this feature

**Comment by Rishabh-malhotraa** [comment]
@starkkumarkk1611 If you haven't started coding this, maybe take a similar issue #52, which adds the feature to load cses problems instead of this one as @ujju20 is also working on this.

**Comment by starkkumarkk1611** [comment]
> @starkkumarkk1611 If you haven't started coding this, maybe take a similar issue #52, which adds the feature to load cses problems instead of this one as @ujju20 is also working on this.

thanks I will work on issue #52,
